### PR TITLE
New resource: azurerm_mssql_managed_instance

### PR DESCRIPTION
**Pull request by manicminer**
* **New Resource:** `azurerm_mssql_managed_instance`

Based off: #15207
Includes: #14950
Depends on: https://github.com/hashicorp/go-azure-helpers/pull/102

Updates go-azure-helpers to v0.25.0

### CODE REVIEW

File: internal/services/mssql/mssql_managed_instance_resource.go [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+			ForceNew:     true,
+		},
+
+		"dns_zone_partner_id": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ValidateFunc: azure.ValidateResourceID,
+		},
+
+		"identity": identity.SystemAssigned{}.Schema(),
```
Comment (tombuildsstuff) [inline internal/services/mssql/mssql_managed_instance_resource.go:191]: can we update this to use the Common Schema and ExpandSystemAssigned / FlattenSystemAssigned packages from `go-azure-helpers`? I'll be removing this package shortly

**Comment by katbyte** [comment]
you might want to include the changes from https://github.com/hashicorp/terraform-provider-azurerm/pull/14950 ?

**Review by katbyte — APPROVED** [review]
aside from changes discussed on slack LGTM 🧊

**Review by katbyte — APPROVED** [review]
LGTM 🤖

**Pull request merged**